patterns/guaranteed_receiver_nack.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"solace.dev/go/messaging"
+	"solace.dev/go/messaging/pkg/solace/config"
+	"solace.dev/go/messaging/pkg/solace/message"
+	"solace.dev/go/messaging/pkg/solace/resource"
+)
+
+func getEnv(key, def string) string {
+	if val, ok := os.LookupEnv(key); ok {
+		return val
+	}
+	return def
+}
+
+func main() {
+	// logging.SetLogLevel(logging.LogLevelInfo)
+
+	// Configuration parameters
+	brokerConfig := config.ServicePropertyMap{
+		config.TransportLayerPropertyHost:                getEnv("SOLACE_HOST", "tcp://localhost:55555,tcp://localhost:55554"),
+		config.ServicePropertyVPNName:                    getEnv("SOLACE_VPN", "default"),
+		config.AuthenticationPropertySchemeBasicPassword: getEnv("SOLACE_PASSWORD", "default"),
+		config.AuthenticationPropertySchemeBasicUserName: getEnv("SOLACE_USERNAME", "default"),
+	}
+
+	messagingService, err := messaging.NewMessagingServiceBuilder().FromConfigurationProvider(brokerConfig).Build()
+
+	if err != nil {
+		panic(err)
+	}
+
+	// Connect to the messaging serice
+	if err := messagingService.Connect(); err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Connected to the broker? ", messagingService.IsConnected())
+
+	if err != nil {
+		panic(err)
+	}
+
+	queueName := "durable-queue"
+	durableExclusiveQueue := resource.QueueDurableExclusive(queueName)
+
+	// Build a Gauranteed message receiver with NACK support and bind to the given queue
+	// Set the required message settlement outcome(s) on the flow using the WithRequiredMessageOutcomeSupport() builder method
+	persistentReceiver, err := messagingService.CreatePersistentMessageReceiverBuilder().
+		WithMessageClientAcknowledgement().
+		// Add message settlement outcomes support on the created Flow here (Failed and Reject for NACK(ing) messages)
+		WithRequiredMessageOutcomeSupport(config.PersistentReceiverFailedOutcome, config.PersistentReceiverRejectedOutcome).
+		Build(durableExclusiveQueue)
+
+	// // Build a Gauranteed message receiver with NACK support and bind to the given queue
+	// // Set the required message settlement outcome(s) on the flow using the configuration provider
+	// persistentReceiver, err := messagingService.CreatePersistentMessageReceiverBuilder().
+	// 	WithMessageClientAcknowledgement().
```

Comment:
@oodigie , We should have a separate sample function or module showing how to configure each one instead of leaving them in comments.

Reply:
Okay. I have made the update